Bottom: 3aa560fa2c7ab0f9eb3ae85b95d01ecf9973ce72
Top:    57c49bb54c28ffd41da0b641c52c54a95e397ab5
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-22 10:32:43 +0800

Refresh of autofs-5.0.5-add-master-read-wait-option.patch

---

diff --git a/CHANGELOG b/CHANGELOG
index 08338b5..b43ad57 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -42,6 +42,7 @@
 - fix submount tree not all expiring.
 - add enable sloppy mount option to configure.
 - wait for master map to be available at start.
+- add master read wait option.
 
 25/07/2012 autofs-5.0.7
 =======================
diff --git a/daemon/automount.c b/daemon/automount.c
index 8c497c9..e8a3984 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1955,9 +1955,10 @@ int main(int argc, char *argv[])
 	unsigned ghost, logging, daemon_check;
 	unsigned dumpmaps, foreground, have_global_options;
 	unsigned master_read;
+	int master_wait;
 	time_t timeout;
 	struct rlimit rlim;
-	const char *options = "+hp:t:vmdD:fVrO:l:n:CF";
+	const char *options = "+hp:t:vmdD:fVrO:l:n:CFM:";
 	static const struct option long_options[] = {
 		{"help", 0, 0, 'h'},
 		{"pid-file", 1, 0, 'p'},
@@ -1974,6 +1975,7 @@ int main(int argc, char *argv[])
 		{"set-log-priority", 1, 0, 'l'},
 		{"dont-check-daemon", 0, 0, 'C'},
 		{"force", 0, 0, 'F'},
+		{"master-wait", 1, 0, 'M'},
 		{0, 0, 0, 0}
 	};
 
@@ -1994,6 +1996,7 @@ int main(int argc, char *argv[])
 	nfs_mount_uses_string_options = check_nfs_mount_version(&vers, &check);
 
 	kpkt_len = get_kpkt_len();
+	master_wait = defaults_get_master_wait();
 	timeout = defaults_get_timeout();
 	ghost = defaults_get_browse_mode();
 	logging = defaults_get_logging();
@@ -2053,6 +2056,10 @@ int main(int argc, char *argv[])
 			dumpmaps = 1;
 			break;
 
+		case 'M':
+			master_wait = getnumopt(optarg, opt);
+			break;
+
 		case 'O':
 			if (!have_global_options) {
 				global_options = strdup(optarg);
diff --git a/include/defaults.h b/include/defaults.h
index 871e14b..1181d25 100644
--- a/include/defaults.h
+++ b/include/defaults.h
@@ -23,6 +23,7 @@
 #define DEFAULT_MASTER_MAP_NAME	"auto.master"
 
 #define DEFAULT_TIMEOUT			600
+#define DEFAULT_MASTER_WAIT		-1
 #define DEFAULT_NEGATIVE_TIMEOUT	60
 #define DEFAULT_MOUNT_WAIT		-1
 #define DEFAULT_UMOUNT_WAIT		12
@@ -59,6 +60,7 @@ unsigned int defaults_read_config(unsigned int);
 const char *defaults_get_master_map(void);
 int defaults_master_set(void);
 unsigned int defaults_get_timeout(void);
+int defaults_get_master_wait(void);
 unsigned int defaults_get_negative_timeout(void);
 unsigned int defaults_get_browse_mode(void);
 unsigned int defaults_get_logging(void);
diff --git a/lib/defaults.c b/lib/defaults.c
index ae1162f..ed521b7 100644
--- a/lib/defaults.c
+++ b/lib/defaults.c
@@ -32,6 +32,7 @@
 #define ENV_NAME_MASTER_MAP		"MASTER_MAP_NAME"
 
 #define ENV_NAME_TIMEOUT		"TIMEOUT"
+#define ENV_NAME_MASTER_WAIT		"MASTER_WAIT"
 #define ENV_NAME_NEGATIVE_TIMEOUT	"NEGATIVE_TIMEOUT"
 #define ENV_NAME_BROWSE_MODE		"BROWSE_MODE"
 #define ENV_NAME_LOGGING		"LOGGING"
@@ -516,6 +517,7 @@ unsigned int defaults_read_config(unsigned int to_syslog)
 
 		if (check_set_config_value(key, ENV_NAME_MASTER_MAP, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_TIMEOUT, value, to_syslog) ||
+		    check_set_config_value(key, ENV_NAME_MASTER_WAIT, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_NEGATIVE_TIMEOUT, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_BROWSE_MODE, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_LOGGING, value, to_syslog) ||
@@ -583,6 +585,17 @@ unsigned int defaults_get_timeout(void)
 	return (unsigned int) timeout;
 }
 
+int defaults_get_master_wait(void)
+{
+	long wait;
+
+	wait = get_env_number(ENV_NAME_MASTER_WAIT);
+	if (wait < 0)
+		wait = DEFAULT_MASTER_WAIT;
+
+	return (int) wait;
+}
+
 unsigned int defaults_get_negative_timeout(void)
 {
 	long n_timeout;
diff --git a/man/auto.master.5.in b/man/auto.master.5.in
index c552e56..091f517 100644
--- a/man/auto.master.5.in
+++ b/man/auto.master.5.in
@@ -211,6 +211,11 @@ They are:
 .B TIMEOUT
 sets the default mount timeout (program default 600).
 .TP
+.B MASTER_WAIT
+sets the default maximum time to wait for the master map to become
+available if it cannot be read at program start (program default -1,
+wait forever).
+.TP
 .B NEGATIVE_TIMEOUT
 Set the default timeout for caching failed key lookups (program default
 60). If the equivalent command line option is given it will override this
diff --git a/man/automount.8 b/man/automount.8
index 7cc32c5..fec2905 100644
--- a/man/automount.8
+++ b/man/automount.8
@@ -34,6 +34,10 @@ Set the global minimum timeout, in seconds, until directories
 are unmounted. The default is 10 minutes. Setting the timeout
 to zero disables umounts completely.
 .TP
+.I "\-M, \-\-master-wait"
+Set the maximum time to wait for the master map to become available
+if it cannot be read at program start.
+.TP
 .I "\-n <seconds>, \-\-negative\-timeout <seconds>"
 Set the default timeout for caching failed key lookups. The default is 60 seconds.
 .TP
diff --git a/redhat/autofs.sysconfig.in b/redhat/autofs.sysconfig.in
index a8992c4..b9d6eef 100644
--- a/redhat/autofs.sysconfig.in
+++ b/redhat/autofs.sysconfig.in
@@ -9,6 +9,13 @@
 #
 TIMEOUT=300
 #
+# MASTER_WAIT - set the default maximum time to wait for the
+# 		master map to become available if it cannot
+# 		be read at program start (default -1, wait
+# 		forever).
+#
+#MASTER_WAIT=-1
+#
 # NEGATIVE_TIMEOUT - set the default negative timeout for
 # 		     failed mount attempts (default 60).
 #
diff --git a/samples/autofs.conf.default.in b/samples/autofs.conf.default.in
index 1da89cf..805b7f3 100644
--- a/samples/autofs.conf.default.in
+++ b/samples/autofs.conf.default.in
@@ -9,6 +9,13 @@
 #
 TIMEOUT=300
 #
+# MASTER_WAIT - set the default maximum time to wait for the
+# 		master map to become available if it cannot
+#		be read at program start (default -1, wait
+#		forever).
+#
+#MASTER_WAIT=-1
+#
 # NEGATIVE_TIMEOUT - set the default negative timeout for
 # 		     failed mount attempts (default 60).
 #
